### PR TITLE
DOC-5694 Prepared Statements

### PR DESCRIPTION
**Pull request by RichardSmedley**

### CODE REVIEW

File: modules/concept-docs/pages/n1ql-query.adoc [hunk 1]
```diff
@@ -16,6 +16,38 @@ Below, we fill in some of the gaps between reference and rolling-up-your-sleeves
...
+
+To indicate that an SDK should optimize a query, the `adhoc()` parameter should be set to false.
+When a query is not ad hoc, the SDK will fetch the plan (if it does not already have it).
+Do not turn off the `adhoc` flag for each query since only a finite number of query plans (currently 5000) can be stored in the SDK.
+
+[source,php]
+----
+$query = 'SELECT x.* FROM `travel-sample` x WHERE x.`type`="hotel" LIMIT 10';
+$opts = new adhoc();
+$opts->adhoc(adhoc::false);
```
Comment (avsej) [inline modules/concept-docs/pages/n1ql-query.adoc]: should be just

```php
$opts->adhoc(false);
```

File: modules/concept-docs/pages/n1ql-query.adoc [hunk 2]
```diff
@@ -16,6 +16,38 @@ Below, we fill in some of the gaps between reference and rolling-up-your-sleeves
...
+When issuing the query again, the SDK will check to see if a plan exists in its cache, and will send the plan to the server.
+
+To indicate that an SDK should optimize a query, the `adhoc()` parameter should be set to false.
+When a query is not ad hoc, the SDK will fetch the plan (if it does not already have it).
+Do not turn off the `adhoc` flag for each query since only a finite number of query plans (currently 5000) can be stored in the SDK.
+
+[source,php]
+----
+$query = 'SELECT x.* FROM `travel-sample` x WHERE x.`type`="hotel" LIMIT 10';
+$opts = new adhoc();
```
Comment (avsej) [inline modules/concept-docs/pages/n1ql-query.adoc]: should be

```php
$opts = new \Couchbase\QueryOptions();
```

File: modules/concept-docs/pages/n1ql-query.adoc (hunk 1 above)
Comment (RichardSmedley) [reply modules/concept-docs/pages/n1ql-query.adoc]: changed :+1:

File: modules/concept-docs/pages/n1ql-query.adoc (hunk 2 above)
Comment (RichardSmedley) [reply modules/concept-docs/pages/n1ql-query.adoc]: Changed

**Review by avsej — APPROVED** [review]

**Pull request merged**